x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -322,28 +322,32 @@ const formatAlertResult = <AlertResult>(
     alertResult;
   const noDataValue = i18n.translate(
     'xpack.infra.metrics.alerting.threshold.noDataFormattedValue',
-    {
-      defaultMessage: '[NO DATA]',
-    }
+    { defaultMessage: '[NO DATA]' }
   );
-  if (!metric.endsWith('.pct'))
+  const thresholdToFormat = useWarningThreshold ? warningThreshold! : threshold;
+  const comparatorToUse = useWarningThreshold ? warningComparator! : comparator;
+
+  if (!metric.endsWith('.pct')) {
+    const formatter = createFormatter('highPrecision');
```

Comment:
💬 I'm using the `highPrecision` formatter since the default number formatter will roundup after one decimal, e.g. `0.84` will become `0.8`

Reply:
Did you use `0.1` for the warning threshold value? 
If you used for example a threshold of `0.15` without `highPrecision`, it would format the reason as `Alert when > 0.2` (0.15 is rounded to 0.2).

Reply:
@kdelemme, the formatter is for the metric current value -in my example `0.10083`, and not for the threshold. 😊 
The `Alert when > 0.1.`  (threshold) is not affected when you change the formatter from `highPrecision` to `number`

Reply:
Checking the code, it seems the formatter is used for both, the current metric current value, and the threshold. But it only affects the metric current value from my tests.

Reply:
Yes the formatter is also used for the threshold value (either alert threshold or warning threshold): [L336-338](https://github.com/elastic/kibana/pull/138138/files/c8d3831b2dfd1a74db9760c8515508eb03c3844e#diff-62ce6361905884966c5c2da1bfa315145e5809dfc3717d543361d91d779f493fR336-R338)

I guess you see the same result for your threshold value because you kept `0.1` in the alert definition which yield the same result for number and highPrecision. Trying with `0.25` would yield `0.3` (and `0.25`) for number (and highPrecision respectively).

I don't mind keeping the `number` formatter, but it might be slightly misleading if the user uses more than one decimal to define the threshold values.